Solutions/GitLab/Analytic Rules/GitLab_ExternalUser.yaml

```diff
@@ -0,0 +1,44 @@
+id: c1544d8f-cbbd-4e35-8d32-5b9312279833
+name: GitLab - External User Added to GitLab
+description: |
+  'This queries GitLab Application logs to list external user accounts (i.e.: account not in allow-listed domains) which have been added to GitLab users.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes: 
+      - Syslog
+queryFrequency: 1h
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Persistence
+relevantTechniques:
+  - T1136
+query: |
+  // List of allow-listed domains
+  let allowedDomain = pack_array("mydomain.com");
+  GitLabAudit
+  | where AddAction == "user"
+  | project AuthorName, IPAddress, UserAdded = EntityPath
+  | join (GitLabApp 
+  | where Message contains "User" and Message contains " was created" 
+  | parse kind=regex Message with EventTime ": User \"" Username "\"" EmailAddress " was created"
+  | project UserAdded = tostring(Username), EmailAddress = substring(EmailAddress,2,strlen(EmailAddress)-3)) on UserAdded
```

Comment:
Why rename to UserAdded, when you can label the field in the parse from line 26?